app/controllers/UserProfileController.java

```diff
@@ -59,4 +70,20 @@ public Result updateUserProfile() throws Exception {
         
         return okResult("Profile updated.");
     }
+    
+    public Result createExternalIdentifier() throws Exception {
+        UserSession session = getAuthenticatedSession();
+
+        ExternalIdentifier externalId = ExternalIdentifier.fromJson(requestToJSON(request()));
+        
+        // TODO: An annotation-based validator would make these trivial validations 
+        // easier to factor out.
+        if (isBlank(externalId.getIdentifier())) {
```

Comment:
Can we push this check to the Service? I know it's possible to test this through the Controller (because that's what you did), but I feel it's cleaner, more testable, and more consistent to have validation in the Service.